include/geode/mesh/core/detail/facet_storage.h

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2019 Geode-solutions
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#pragma once
+
+#include <geode/basic/common.h>
+#include <unordered_map>
+
+#include <geode/basic/attribute.h>
+#include <geode/basic/attribute_manager.h>
+#include <geode/basic/range.h>
+#include <geode/mesh/builder/detail/mapping_after_deletion.h>
+#include <geode/mesh/core/detail/vertex_cycle.h>
+
+namespace geode
+{
+    namespace detail
+    {
+        class FacetStorage
+        {
+        protected:
+            FacetStorage()
+                : counter_(
+                    facet_attribute_manager_
+                        .template find_or_create_attribute< VariableAttribute,
+                            index_t >( "counter", 1 ) )
```

Comment:
Why default is 1 and not 0?

Reply:
Two solutions : set 0 by default and do counter++ if facet is created in the add_facet method, or set 1 by default and not do counter++ if facet is created in the add_facet method. What do you suggest?

Reply:
Ok for 1 for now